tests/sentry/event_manager/test_save_aggregate.py

```diff
@@ -48,27 +53,30 @@ def atomic(*args, **kwds):
     return_values = []
 
     def save_event():
-        data = {"timestamp": time.time()}
-        evt = Event(
-            default_project.id,
-            "89aeed6a472e4c5fb992d14df4d7e1b6",
-            data=data,
-        )
-        ret = _save_aggregate(
-            evt,
-            hashes=CalculatedHashes(
-                hashes=["a" * 32, "b" * 32],
-                hierarchical_hashes=[],
-                tree_labels=[],
-            ),
-            release=None,
-            metadata={},
-            received_timestamp=0,
-            level=10,
-            culprit="",
-        )
-        assert ret is not None
-        return_values.append(ret)
+        try:
+            data = {"timestamp": time.time()}
+            evt = Event(
+                default_project.id,
+                "89aeed6a472e4c5fb992d14df4d7e1b6",
+                data=data,
+            )
+            ret = _save_aggregate(
+                evt,
+                hashes=CalculatedHashes(
+                    hashes=["a" * 32, "b" * 32],
+                    hierarchical_hashes=[],
+                    tree_labels=[],
+                ),
+                release=None,
+                metadata={},
+                received_timestamp=0,
+                level=10,
+                culprit="",
+            )
+            assert ret is not None
+            return_values.append(ret)
+        finally:
+            transaction.get_connection(router.db_for_write(GroupHash)).close()
```

Comment:
Because otherwise the connection remains open until python happens to garbage collect it -- django creates new connections per thread but doesn't close them automatically.  Then, randomly, tests will fail during teardown thanks to the open connection.

Closing ensures proper cleanup between test runs.